code/modules/projectiles/projectile.dm

```diff
@@ -684,9 +684,13 @@
 	return TRUE
 
 /obj/item/projectile/Process()
-	var/first_step = TRUE
-
+	var/first_step = 1
+	var/i = 0
 	spawn while(src && src.loc)
+		if (++i > 512)
```

Comment:
Why 512?

Reply:
Worked for Onyx, a direct copypaste from their fix